mlflow/models/evaluation/base.py

```diff
@@ -1735,57 +1736,107 @@ def pred_sample(eval_df, _builtin_metrics, _artifacts_dir):
 
     # Use specified model_id if provided, otherwise use derived model_id
     model_id = specified_model_id if specified_model_id is not None else model_id
+    # If none of the model_id and model is specified, use the active model_id
+    if model_id is None:
+        model_id = mlflow.get_active_model_id()
+    # model_id could be None
+    with _maybe_set_active_model_id(model_id=model_id):
```

Comment:
can we use `nullcontext` instead of creating `_maybe_set_active_model_id`?

https://docs.python.org/3.7/library/contextlib.html#contextlib.nullcontext

Reply:
```suggestion
    with _set_active_model_id(model_id=model_id) if model_id else nullcontext()
```